Stats: Simplify highlights data setup on Ads page

a8ck3n opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #87375

## Proposed Changes

Adds SVG property to initial setup so we can simplify things in the mobile highlights code path. No UI updates. Should still look like this:

<img width="439" alt="SCR-20240405-pckv" src="https://github.com/Automattic/wp-calypso/assets/40267301/b2c8c8c0-18c4-4e71-bd96-5f6e0818f5ea">

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Open the Calypso Live branch.
* Navigate to the **Stats → Ads** page.
* Use the browser tools to test at different screen sizes. iPad mini and larger should get the standard Desktop UI with horizontal scrolling while smaller screens should get the vertically stacked cards.
* Confirm **Traffic**, **Insights**, & **Subscriber** pages all continue to behave correctly.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

a8ck3n committed:
Add SVG data to setup stage.

a8ck3n committed:
Update mobile code path to use a loop for setup.

a8ck3n committed:
Comment++

a8ck3n committed:
Merge branch 'trunk' into update/stats-ads-page-mobile-highlights-tweak-2

kangzj commented on client/my-sites/stats/wordads/highlights-section.jsx:
```diff
@@ -15,6 +15,9 @@ function getAmountAsFormattedString( amount, stripZeros ) {
 }
 
 function useHighlights( earnings ) {
```
IMHO It's very important to keep the interface consistent, rather than to simplify things in the mobile highlights code path, for the sake of easier adoption. If it's possible to handle the differences in one place rather than everywhere, then it's a win. Thoughts @dognose24?

a8ck3n replied:
I'm not sure I understand your meaning here. Also related, some previous feedback from Dognose here:

https://github.com/Automattic/wp-calypso/pull/89360#pullrequestreview-1989758920

Are you recommending that I stick with sending along the `Icon` component and leave the SVG value out? What do you think about the hidden dependency in the `CountCard` component? I was thinking a good next step would be to update `CountCard` so it imports `Icon` and uses it directly.

It would be a small/quick change. I'm good with either approach though.

a8ck3n replied:
I updated the `CountCard` to make the `Icon` dependency explicit. This also makes the interface consistent across mobile and Desktop.

a8ck3n committed:
Update CountCard to make Icon dependecy explicit.

On the assumption that React.ReactNode and JSX.Element are too broad. At least this way, the dependency is exposed directly in the file.

a8ck3n committed:
Update CountCard usage to send SVG data.

Makes things consistent across mobile and Desktop.
